### PR TITLE
fix: Alert on detachAddress

### PR DESCRIPTION
**Pull request by Ambrdctr**

### CODE REVIEW

File: Mail/Views/Alerts/DetachMailboxConfirmationView.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    private func detach() {
+        @InjectService var matomo: MatomoUtils
+        matomo.track(eventWithCategory: .invalidPasswordMailbox, name: "detachMailboxConfirm")
+
+        Task {
+            do {
+                try await AccountManager.instance.detachMailbox(mailbox: mailbox)
+                await (window?.windowScene?.delegate as? SceneDelegate)?.showMainView()
+            } catch {
+                print(error)
```
Comment (valentinperignon) [inline Mail/Views/Alerts/DetachMailboxConfirmationView.swift:49]: Maybe we should use `tryOrDisplayError` here?

File: Mail/Views/Unavailable Mailbox/UpdateMailboxPasswordView.swift [hunk 2]
```diff
@@ -30,6 +30,7 @@ struct UpdateMailboxPasswordView: View {
     @State private var updatedMailboxPassword = ""
     @State private var isShowingError = false
     @State private var isLoading = false
+    @State private var showDetachMailboxAlert = false
 
     private var disableButton: Bool {
```
Comment (valentinperignon) [inline Mail/Views/Unavailable Mailbox/UpdateMailboxPasswordView.swift:35]: Is this variable used somewhere?

**Review by valentinperignon — COMMENTED** [review]
Great 👍 Could you update DetachMailboxConfirmationView with the updated design?

File: Mail/Views/Unavailable Mailbox/UpdateMailboxPasswordView.swift [hunk 3]
```diff
@@ -30,6 +30,7 @@ struct UpdateMailboxPasswordView: View {
     @State private var updatedMailboxPassword = ""
     @State private var isShowingError = false
     @State private var isLoading = false
+    @State private var showDetachMailboxAlert = false
```
Comment (PhilippeWeidmann) [inline Mail/Views/Unavailable Mailbox/UpdateMailboxPasswordView.swift:33]: Could we start enforcing using `is` at the beginning of boolean properties that show views ?
And always include View in the property.
-> isShowDetachMailboxAlertView

File: Mail/Views/Unavailable Mailbox/UpdateMailboxPasswordView.swift [hunk 4]
```diff
@@ -30,13 +30,14 @@ struct UpdateMailboxPasswordView: View {
     @State private var updatedMailboxPassword = ""
     @State private var isShowingError = false
     @State private var isLoading = false
+    @State private var isShowDetachMailboxAlertView = false
```
Comment (valentinperignon) [inline Mail/Views/Unavailable Mailbox/UpdateMailboxPasswordView.swift:33]:
```suggestion
    @State private var isShowingDetachMailboxAlertView = false
```

File: Mail/Views/Alerts/DetachMailboxConfirmationView.swift [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+struct DetachMailboxConfirmationView: View {
+    @Environment(\.window) private var window
+
+    let mailbox: Mailbox
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 24) {
+            Text(MailResourcesStrings.Localizable.buttonDetachMailbox)
+                .textStyle(.bodyMedium)
+            Text(MailResourcesStrings.Localizable.popupDetachMailboxDescription(mailbox.email))
```
Comment (valentinperignon) [inline Mail/Views/Alerts/DetachMailboxConfirmationView.swift:34]: Could we use an `AttributedString` so the mail adress can have a bodyMedium font?

File: Mail/Views/Alerts/DetachMailboxConfirmationView.swift [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+import SwiftUI
+
+struct DetachMailboxConfirmationView: View {
+    @EnvironmentObject private var navigationState: NavigationState
+
+    let mailbox: Mailbox
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: 24) {
+            Text(MailResourcesStrings.Localizable.buttonDetachMailbox)
```
Comment (valentinperignon) [inline Mail/Views/Alerts/DetachMailboxConfirmationView.swift:32]: We must use `popupDetachMailboxTitle`

File: Mail/Views/Unavailable Mailbox/UpdateMailboxPasswordView.swift [hunk 7]
```diff
@@ -47,7 +48,7 @@ struct UpdateMailboxPasswordView: View {
                     .textStyle(.bodySecondary)
                 MailButton(label: MailResourcesStrings.Localizable.buttonDetachMailbox) {
                     matomo.track(eventWithCategory: .invalidPasswordMailbox, name: "detachMailbox")
-                    detachAddress()
+                    isShowingDetachMailboxAlertView.toggle()
```
Comment (valentinperignon) [inline Mail/Views/Unavailable Mailbox/UpdateMailboxPasswordView.swift:51]:
```suggestion
                    isShowingDetachMailboxAlertView = true
```

**Pull request merged**